### PR TITLE
added Declarative table to model view controls

### PR DESCRIPTION
**Pull request by llali**

### CODE REVIEW

File: src/sql/parts/modelComponents/components.contribution.ts [hunk 1]
```diff
@@ -10,7 +10,11 @@ import GroupContainer from './groupContainer.component';
 import CardComponent from './card.component';
 import InputBoxComponent from './inputbox.component';
 import DropDownComponent from './dropdown.component';
+<<<<<<< HEAD
```
Comment (AbbiePetcht) [inline src/sql/parts/modelComponents/components.contribution.ts]: can you resolve the conflict?

File: samples/sqlservices/src/controllers/mainController.ts [hunk 2]
```diff
@@ -188,6 +188,45 @@ export default class MainController implements vscode.Disposable {
...
+						width: '100px',
+						isReadOnly: false
+					}, {
+						displayName: 'Column 3',
+						valueType: sqlops.DeclarativeDataType.boolean,
+						width: '20px',
+						isReadOnly: false
+					}, {
+						displayName: 'Column 4',
+						valueType: sqlops.DeclarativeDataType.category,
```
Comment (AbbiePetcht) [inline samples/sqlservices/src/controllers/mainController.ts]: >valueType [](start = 6, length = 9)

Should create a work item to support input validation

File: src/sql/parts/modelComponents/declarativeTable.component.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+	category = 'category',
+	boolean = 'boolean'
+}
+
+@Component({
+	selector: 'modelview-declarativeTable',
+	template: `
+	<table role=grid aria-labelledby="ID_REF" #container *ngIf="columns" class="declarative-table">
+		<ng-container *ngFor="let column of columns;let h = index">
+			<th class="declarative-table-header" tabindex="-1" role="button" aria-sort="none">{{column.displayName}}</th>
```
Comment (AbbiePetcht) [inline src/sql/parts/modelComponents/declarativeTable.component.ts]: >role="button" [](start = 54, length = 13)

Header's role is button? Is it right?

File: samples/sqlservices/src/controllers/mainController.ts (hunk 2 above)
Comment (llali) [reply samples/sqlservices/src/controllers/mainController.ts]: I will

---
In reply to: [194467324](https://github.com/Microsoft/sqlopsstudio/pull/1593#discussion_r194467324) [](ancestors = 194467324)

File: src/sql/parts/dashboard/dashboard.module.ts [hunk 4]
```diff
@@ -52,11 +52,16 @@ import { DashboardControlHostContainer } from 'sql/parts/dashboard/containers/da
...
+import { SelectBox } from 'sql/base/browser/ui/selectBox/selectBox.component';
+import { InputBox } from 'sql/base/browser/ui/inputBox/inputBox.component';
 
 let baseComponents = [DashboardHomeContainer, DashboardComponent, DashboardWidgetWrapper, DashboardWebviewContainer,
 	DashboardWidgetContainer, DashboardGridContainer, DashboardErrorContainer, DashboardNavSection, ModelViewContent, WebviewContent, WidgetContent,
 	ComponentHostDirective, BreadcrumbComponent, ControlHostContent, DashboardControlHostContainer,
-	JobsViewComponent, AgentViewComponent, JobHistoryComponent, JobStepsViewComponent, DashboardModelViewContainer, ModelComponentWrapper];
+	JobsViewComponent, AgentViewComponent, JobHistoryComponent, JobStepsViewComponent, DashboardModelViewContainer, ModelComponentWrapper, Checkbox,
+	SelectBox,
+	InputBox,];
```
[the remark this answers is not in the document]
Comment (llali) [reply src/sql/parts/dashboard/dashboard.module.ts]: it was failing becuase model view module requires those three now

---
In reply to: [194469079](https://github.com/Microsoft/sqlopsstudio/pull/1593#discussion_r194469079) [](ancestors = 194469079)

File: src/sql/parts/modelComponents/declarativeTable.component.ts (hunk 3 above)
Comment (llali) [reply src/sql/parts/modelComponents/declarativeTable.component.ts]: I work on accessibility in another PR

---
In reply to: [194470319](https://github.com/Microsoft/sqlopsstudio/pull/1593#discussion_r194470319) [](ancestors = 194470319)

**Pull request merged**